### DISCUSSION TITLE
System time check

### DISCUSSION THREAD

**Discussion by digitalfredy** [ente-io/ente#498] [Enhancements]
**Is your feature request related to a problem? Please describe.**

I deployed auth for 35 users sending to them and I had hard time with one of them:

- I pre-configured 2FA per each user using auth and exporting the data using encryption
- Users imported the json file in _auth_ and decrypt it using the password
- The user with the issue was not capable to login, wrong credentials while inserting the 2FA provided by _auth_

After a looong troubleshooting I found this user had his clock few minutes in advance and this why the 2FA was not working.

**Describe the solution you'd like**
Will be great if _auth_ check the system configuration and inform the end user if his clock is not configured properly. 

If the app can not check the system clock setup can check the time from an internet time server and validate if is the same provided by the system

**Describe alternatives you've considered**
If can not be automated a message is an alternative but is not good idea to disturb 95% of the users with this message addressed to the 5%

**Additional context**
User device: Android

**Comment by chrissv** [comment]
Question: does ente auth connect to the internet for any other reason under normal conditions (i.e. already logged in)?

**Reply by ua741** [reply]
> does ente auth connect to the internet for any other reason under normal conditions (i.e. already logged in)?
We do connect to internet to see if there are any changes that might been made on another device.

**Comment by ua741** [comment]
https://help.ente.io/auth/faq/#why-do-the-desktop-and-mobile-apps-display-different-codes

> Newer Ente Auth clients (upcoming 4.4.0+) will automatically try to correct for incorrect system time, so you should be seeing correct codes even if your system time is out of sync. However, this automatic correction will not work if you're using Ente Auth in offline mode.
> 
> If you've recently changed your system time and the codes are still incorrect, try to refresh / restart the app if needed.
